sorc/build_all.sh

```diff
@@ -105,170 +107,190 @@ ERRSCRIPT=${ERRSCRIPT:-'eval [[ $err = 0 ]]'}
+         fi
+      fi
+      if [[ -n "${build_jobs[ww3prepost]+0}" && -z "${build_ids[ww3prepost]+0}" ]]; then
+         if [[ ${_build_job_max} -ge $(( build_jobs['ww3prepost'] + procs_in_use )) ]]; then
+            ./build_ww3prepost.sh -j "${build_jobs[ww3prepost]}" "${_verbose_opt}" "${_build_ufs_opt}" > \
+               "${logs_dir}/build_ww3prepost.log" 2>&1 &
+            build_ids["ww3prepost"]=$!
+            echo "Starting build_ww3prepost.sh"
+            procs_in_use=$(( procs_in_use + build_jobs['ww3prepost'] ))
+         fi
+      fi
+      if [[ -n "${build_jobs[gsi_monitor]+0}" && -z "${build_ids[gsi_monitor]+0}" ]]; then
+         if [[ ${_build_job_max} -ge $(( build_jobs['gsi_monitor'] + procs_in_use )) ]]; then
+            ./build_gsi_monitor.sh -j "${build_jobs[gsi_monitor]}" "${_verbose_opt}" > \
+               "${logs_dir}/build_gsi_monitor.log" 2>&1 &
+            build_ids["gsi_monitor"]=$!
+            echo "Starting build_gsi_monitor.sh"
+            procs_in_use=$(( procs_in_use + build_jobs['gsi_monitor'] ))
+         fi
+      fi
+      if [[ -n "${build_jobs[gfs_utils]+0}" && -z "${build_ids[gfs_utils]+0}" ]]; then
+         if [[ ${_build_job_max} -ge $(( build_jobs['gfs_utils'] + procs_in_use )) ]]; then
+            ./build_gfs_utils.sh -j "${build_jobs[gfs_utils]}" "${_verbose_opt}" > \
+               "${logs_dir}/build_gfs_utils.log" 2>&1 &
+            build_ids["gfs_utils"]=$!
+            echo "Starting build_gfs_utils.sh"
+            procs_in_use=$(( procs_in_use + build_jobs['gfs_utils'] ))
+         fi
+      fi
+   fi
+
+   # Check if all builds have completed
+   # Also recalculate how many processors are in use to account for completed builds
+   builds_started=0
+   procs_in_use=0
+   for build in "${!build_jobs[@]}"; do
+      if [[ -n "${build_ids[${build}]+0}" ]]; then
+         builds_started=$(( builds_started + 1))
+         # Calculate how many processors are in use
+         if ps -p "${build_ids[${build}]}" > /dev/null; then
+            procs_in_use=$(( procs_in_use + build_jobs["${build}"] ))
+         fi
+      fi
+   done
+
+   sleep 5s
+done
+
+# Wait for all jobs to complete and check return statuses
+errs=0
+while [[ ${#build_jobs[@]} -gt 0 ]]; do
+   for build in "${!build_jobs[@]}"; do
+      # Test if each job is complete and if so, notify and remove from the array
+      if [[ -n "${build_ids[${build}]+0}" ]]; then
+         if ! ps -p "${build_ids[${build}]}" > /dev/null; then
+            wait "${build_ids[${build}]}"
+            build_stat=$?
+            errs=$((errs+build_stat))
+            if [[ ${build_stat} == 0 ]]; then
+               echo "${build} completed successfully!"
+            else
+               echo "${build} failed with status ${build_stat}!"
+            fi
+
+            # Remove the completed build from the list of PIDs
+            unset 'build_ids[${build}]'
+            unset 'build_jobs[${build}]'
+            break
```

Comment:
I'm assuming this is the reason for the break here. Would like a comment so no one else wonders why.
```suggestion
            # Removing elements from the loop array can cause issues, so just start the loop over.
            break
```

Reply:
The break was just for testing.  It seems to handle the deletion fine without it, so I have removed it.